covidthailand.py

```diff
@@ -2597,794 +2601,1157 @@ def topprov(df, metricfunc, valuefunc=None, name="Top 5 Provinces", num=5, other
+    #     kind='area',
+    #     figsize=[20, 10],
+    #     colormap=custom_cm(AREA_COLOURS, len(cols), last_colour='lightgrey'),
+    #     title=title
+    # )
+    # ax.legend(legends)
+    # plt.tight_layout()
+    # plt.savefig('outputs/cases_areas_all.png')
 
-    fig, ax = plt.subplots()
-    df.plot(
-        ax=ax,
-        y=cols,
-        kind="area",
-        figsize=[20, 10],
-        colormap=AREA_COLOURS,
-        title=title
-    )
-    ax.legend(legend)
-    plt.tight_layout()
-    plt.savefig("outputs/cases_areas_all.png")
+    cols = rearrange([f'Cases Walkin Area {area}' for area in range(1, 14)],*FIRST_AREAS)
+    # fig, ax = plt.subplots()
+    # df['2021-02-16':].plot(
+    #     ax=ax,
+    #     y=cols,
+    #     kind='area',
+    #     figsize=[20, 10],
+    #     colormap=custom_cm(AREA_COLOURS, len(cols), last_colour='lightgrey'),
+    #     title='Thailand 'Walkin' Covid Cases by health District\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend(AREA_LEGEND_UNKNOWN)
+    # plt.tight_layout()
+    # plt.savefig('outputs/cases_areas_walkins.png')
+    plot_area(df=df, png_prefix='cases_areas_walkins', cols_subset=cols,
+              title='Thailand "Walkin" Covid Cases by Health District', legends=AREA_LEGEND_UNKNOWN,
+              kind='area', stacked=True, percent_fig=False, ma=False, cmap='tab20')
 
 
-    cols = rearrange([f"Cases Walkin Area {area}" for area in range(1, 15)],*FIRST_AREAS)
-    fig, ax = plt.subplots()
-    df["2021-02-16":].plot(
-        ax=ax,
-        y=cols,
-        kind="area",
-        figsize=[20, 10],
-        colormap=AREA_COLOURS,
-        title='Thailand "Walkin" Covid Cases by health District\n'
-        f"Updated: {TODAY().date()}\n"        
-        "djay.github.io/covidthailand"
-    )
-    ax.legend(AREA_LEGEND_UNKNOWN)
-    plt.tight_layout()
-    plt.savefig("outputs/cases_areas_walkins.png")
+    cols = rearrange([f'Cases Proactive Area {area}' for area in range(1, 14)],*FIRST_AREAS)
+    # fig, ax = plt.subplots(figsize=[20, 10],)
+    # df['2021-02-16':].plot.area(
+    #     ax=ax,
+    #     y=cols,
+    #     colormap=custom_cm(AREA_COLOURS, len(cols), last_colour='lightgrey'),
+    #     title='Thailand 'Proactive' Covid Cases by health District\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend(AREA_LEGEND_UNKNOWN)
+    # plt.tight_layout()
+    # plt.savefig('outputs/cases_areas_proactive.png')
+    plot_area(df=df, png_prefix='cases_areas_proactive', cols_subset=cols,
+              title='Thailand "Proactive" Covid Cases by Health District', legends=AREA_LEGEND_UNKNOWN,
+              kind='area', stacked=True, percent_fig=False, ma=False, cmap='tab20')
 
-    cols = rearrange([f"Cases Proactive Area {area}" for area in range(1, 15)],*FIRST_AREAS)
-    fig, ax = plt.subplots(figsize=[20, 10],)
-    df["2021-02-16":].plot.area(
```

Comment:
need to be careful of which time periods I've used. there might be more than _full, _1, _2, _3.